src/nix/host/local.rs

```diff
@@ -149,7 +150,13 @@ impl Local {
 
         let parent_dir = dest_path.parent().unwrap();
         fs::create_dir_all(parent_dir)?;
-        fs::rename(temp_path, dest_path)?;
+
+        if fs::rename(&temp_path, &dest_path).is_err() {
+            // Linux can not rename across different filesystems, try copy-then-remove
```

Comment:
Noticed this when I tried to deploy secrets locally to `/run/keys`, which is a separate tmpfs mount.

Reply:
Nice catch!